src/manifold.cpp

```diff
@@ -374,14 +371,46 @@ size_t Manifold::NumPropVert() const {
 Box Manifold::BoundingBox() const { return GetCsgLeafNode().GetImpl()->bBox_; }
 
 /**
- * Returns the precision of this Manifold's vertices, which tracks the
+ * Returns the epsilon value of this Manifold's vertices, which tracks the
  * approximate rounding error over all the transforms and operations that have
- * led to this state. Any triangles that are colinear within this precision are
- * considered degenerate and removed. This is the value of &epsilon; defining
+ * led to this state. This is the value of &epsilon; defining
  * [&epsilon;-valid](https://github.com/elalish/manifold/wiki/Manifold-Library#definition-of-%CE%B5-valid).
  */
-double Manifold::Precision() const {
-  return GetCsgLeafNode().GetImpl()->precision_;
+double Manifold::GetEpsilon() const {
+  return GetCsgLeafNode().GetImpl()->epsilon_;
+}
+
+Manifold Manifold::SetEpsilon(double epsilon) const {
+  auto impl = std::make_shared<Impl>(*GetCsgLeafNode().GetImpl());
+  auto oldTolerance = impl->tolerance_;
+  impl->SetEpsilon(epsilon);
+  if (impl->tolerance_ > oldTolerance) impl->SimplifyTopology();
```

Comment:
If we moved this logic into the Impl functions could we reduce duplication by having SetEpsilon call SetTolerance?

Reply:
yes